core/src/main/java/com/google/cloud/sql/core/DefaultConnectionInfoCache.java

```diff
@@ -107,6 +104,23 @@ SSLSocket createSslSocket(long timeoutMs) throws IOException {
         getConnectionInfo(timeoutMs).getSslContext().getSocketFactory().createSocket();
   }
 
+  ConnectionMetadata getConnectionMetadata(long timeoutMs) {
```

Comment:
https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/blob/main/core/src/main/java/com/google/cloud/sql/core/Connector.java#L110 could call `instance.getConnectionMetadata(..).getIpAddress()` and the method `getPreferredIp()` could be deleted.

Reply:
Good catch. Fixed.